src/opera_rtc_s1_browse/create_browse.py

```diff
@@ -17,6 +18,28 @@
 s3 = boto3.client('s3')
 
 
+def now() -> datetime.datetime:
+    return datetime.datetime.now()
+
+
+def create_metadata_xml(co_pol_path: Path, browse_path: Path) -> Path:
+    info = gdal.Info(str(co_pol_path), format='json')
+    start_time = datetime.datetime.fromisoformat(info["metadata"][""]["ZERO_DOPPLER_START_TIME"][:-1])
+    end_time = datetime.datetime.fromisoformat(info["metadata"][""]["ZERO_DOPPLER_END_TIME"][:-1])
```

Comment:
The `[""]` threw me off. The empty string is used as a dictionary key?

Reply:
Also, what's the reason for removing the last character with `[:-1]`?

Reply:
Yes, that's how gdalinfo organizes the metadata key/value pairs when you ask for them in json format

```
$ gdalinfo OPERA_L2_RTC-S1_T113-241591-IW2_20240610T110704Z_20240610T154200Z_S1A_30_v1.0_VV.tif
Driver: GTiff/GeoTIFF
Files: OPERA_L2_RTC-S1_T113-241591-IW2_20240610T110704Z_20240610T154200Z_S1A_30_v1.0_VV.tif
Size is 3430, 1614
...
Metadata:
  ABSOLUTE_ORBIT_NUMBER=54260
  ACQUISITION_MODE=IW
  AREA_OR_POINT=Area

```

```
$ gdalinfo -json OPERA_L2_RTC-S1_T113-241591-IW2_20240610T110704Z_20240610T154200Z_S1A_30_v1.0_VV.tif
{
  "description":"OPERA_L2_RTC-S1_T113-241591-IW2_20240610T110704Z_20240610T154200Z_S1A_30_v1.0_VV.tif",
  "driverShortName":"GTiff",
  "driverLongName":"GeoTIFF",
  "files":[
    "OPERA_L2_RTC-S1_T113-241591-IW2_20240610T110704Z_20240610T154200Z_S1A_30_v1.0_VV.tif"
  ],
  "size":[
    3430,
    1614
  ],
  ...
  "metadata":{
    "":{
      "ABSOLUTE_ORBIT_NUMBER":"54260",
      "ACQUISITION_MODE":"IW",
      "AREA_OR_POINT":"Area",
```

Reply:
datetime.fromisoformat() doesn't like `Z` as a timezone identifier:
```
>>> datetime.datetime.fromisoformat('2024-06-14T00:00:00Z')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: Invalid isoformat string: '2024-06-14T00:00:00Z'
>>> datetime.datetime.fromisoformat('2024-06-14T00:00:00')
datetime.datetime(2024, 6, 14, 0, 0)
>>> datetime.datetime.fromisoformat('2024-06-14T00:00:00+00:00')
datetime.datetime(2024, 6, 14, 0, 0, tzinfo=datetime.timezone.utc)
```